locator/tablets.cc

```diff
@@ -435,6 +435,62 @@ load_stats& load_stats::operator+=(const load_stats& s) {
     return *this;
 }
 
+tablet_range_splitter::tablet_range_splitter(schema_ptr schema, const tablet_map& tablets, host_id host, const dht::partition_range_vector& ranges)
+    : _schema(std::move(schema))
+    , _ranges(ranges)
+    , _ranges_it(_ranges.begin())
+{
+    // Filter all tablets and save only those that have a replica on the specified host.
+    for (auto tid = std::optional(tablets.first_tablet()); tid; tid = tablets.next_tablet(*tid)) {
+        const auto& tablet_info = tablets.get_tablet_info(*tid);
+
+        auto replica_it = std::ranges::find_if(tablet_info.replicas, [&] (auto&& r) { return r.host == host; });
+        if (replica_it == tablet_info.replicas.end()) {
+            continue;
+        }
+
+        auto token_range = tablets.get_token_range(*tid);
+        auto range = dht::partition_range(
+                dht::partition_range::bound(dht::ring_position(token_range.start()->value(), dht::ring_position_view::token_bound::end), false),
+                dht::partition_range::bound(dht::ring_position(token_range.end()->value(), dht::ring_position_view::token_bound::end), true));
```

Comment:
Don't we automatic conversion from token_range to partition range? In any case we shouldn't assume anything about inclusiveness/exclusiveness. Better move to a helper, and use interval::transform to let it handle the bounds.

Reply:
maybe that's not trivial because of token_bound. But we shouldn't assume anything about inclusiveness/exclusiveness and shouldn't bury the calculations here.

Reply:
Yes, it is another case of `T` already having inclusiveness built-in, and then there is another `is_inclusive` in the range bound. `transform()` takes the inclusiveness of the bound, but there is no way to communicate the underlying `T`'s inclusiveness.

We should really address this at one point, it is the same situation with `position_in_partition`.
We could add an `inclusiveness_policy` (maybe with a better name) to `interval`, which by default it stores it in the `bound`, but can also use an already existing inclusiveness flag in `T`.

Reply:
Let's separate it to a help, that deals with all edge cases and makes no assumptions, and comes with a small test suite.

Reply:
I think it's not hard - token range to partition range is unambiguous (the other direction is ambiguous because you can return a range that is the smallest superset of the input, or the largest subset of the input, but there's no exact mapping.

Reply:
We already have 

```c++
dht::partition_range to_partition_range(dht::token_range);
```

Reply:
Yes, I implemented this myself, then found out we have it (I missed this comment). Although, the existing one needed a fix.